torch/distributed/fsdp/flat_param.py

```diff
@@ -1139,6 +1169,40 @@ def _use_unsharded_views(self, as_params: bool) -> None:
             else:
                 setattr(module, param_name, prim_param)
 
+    def _use_unsharded_grad_views(self) -> None:
+        """
+        Unflattens the unsharded flattened parameter's gradient by setting the
+        original module parameter variables' gradients to be views into it.
+        """
+        if self.flat_param.grad is None:
+            return
+        self._check_unsharded(self.flat_param.grad)
+        views = self._get_unflat_views(self.flat_param, self.flat_param.grad)
+        for i, (view, (param_name, module, _)) in enumerate(
+            zip(views, self.flat_param._param_infos)
+        ):
+            p_assert(
+                hasattr(module, param_name),
+                f"{self.flat_param._prefixed_param_names[i]} is missing",
+            )
+            param = getattr(module, param_name)
+            param.grad = view
+        for i, (
```

Comment:
This just makes sure that each shared parameter's `.grad` is also populated. Without this, then if we had a model like
```
lin = nn.Linear(5, 5)
lin.weight = lin.bias
```
then only one of `(lin.weight, lin.bias)` would get a `.grad` since only one of them is in accounted for in `_param_infos` / `_params` / etc.